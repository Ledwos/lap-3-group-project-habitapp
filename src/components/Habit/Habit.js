import React, { useState } from 'react'

export default function Habit(props) {


    return (
        <div>
            <p>{props.title}</p>
        </div>
    )
}
