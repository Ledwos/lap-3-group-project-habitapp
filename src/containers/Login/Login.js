import React, { useState } from "react";
import axios from "axios";
import '../App/App.css';

export default function Login() {
  const [user, setUser] = useState({ username: null });
  const [password, setPassword] = useState({ password: null });

  // function handleUser(event) {
  //     setUser({username: event.target.value})
  // }

  // function handlePassword(event) {
  //     setPassword({password: event.target.value})
  // }

  // function handleSubmit(event) {
  //     event.preventDefault();
  //     const form = new FormData();
  //     form.append('user', user);
  //     form.append('password', password);

  //     axios({
  //         method: 'post',
  //         url: 'http://localhost:8081/authentication',
  //         data: form,
  //         headers: {'content-type': 'multipart/form-data; boundary=${form._boundary}' }
  //     })
  //     .then(res => console.log(res))
  //     .catch(error => console.error(error))

  // }

  return (
    <div>
      <h2 className="app_title">Habitapp</h2>
      <div className="login_box_outer">
        <div className="login_box">
          <div className="login_vert_box">
            <form action="http://localhost:8081/authentication" method="POST">
              <div className="form_component">
              <label for="username">Username:</label>
              <input type="text" name="username"></input>
              </div>
              <div className="form_component">
              <label for="password">Password:</label>
              <input type="password" name="password"></input>
              </div>
              <input className="submit_button" type="submit" value="submit"></input>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}
