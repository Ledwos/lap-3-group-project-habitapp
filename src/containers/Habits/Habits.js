import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import Habit from "../../components/Habit/Habit";

export default function Habits(props) {
  const [state, setState] = useState({ habits: [] });
  const id = window.location.search.slice(4);
  useEffect(() => {
    fetch(`http://localhost:8081/users/${id}/habits`)
      .then(res => res.json())
      .then(res => setState({ habits: res }));
    // .then(res => console.log(res));
  });

  function habitMap() {
    return state.habits.map((habit, index) => (
      <Habit
        title={habit.title}
        habitId={habit._id}
        freq_actual={habit.freq_actual}
        freq_goal={habit.freq_goal}
        streak={habit.streak}
        userId={id}
      />
    ));
  }
  return (
    <div>
      <p>
        Add a habit by <Link to={"/add_habit?id=" + id}>clicking here</Link>
      </p>
      <p>This will have a list of habits!</p>
      {state.habits.length > 0 ? habitMap() : <p>didn't render</p>}
    </div>
  );
}
