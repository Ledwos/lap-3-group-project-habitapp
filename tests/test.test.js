import React from 'react';
import {shallow, mount} from 'enzyme';
import Login from '../src/containers/Login/Login';
import Habits from '../src/containers/Habits/Habits';
import AddHabit from '../src/components/Add_habit/Add_habit';
import Habit from '../src/components/Habit/Habit';


describe("Login.js", () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<Login />))

    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it("Should render a h2 element", () => {
        expect(wrapper.find("h2").length).toEqual(1);
    })
})

describe("Habit.js", () => {
    let wrapper;

    beforeEach(() => wrapper = mount(<Habit />))

    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it("Should render a div element", () => {
        expect(wrapper.find("div").length).toEqual(1);
    })

    it("Should render a a element", () => {
        expect(wrapper.find("a").length).toEqual(1);
    })
});

describe("Shallow Habits", () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<Habits />))

    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it("Should render a h2 element", () => {
        expect(wrapper.find("h2").length).toEqual(1);
    })

    it("Should render 3 p elements", () => {
        expect(wrapper.find("p").length).toEqual(3);
    })

})
