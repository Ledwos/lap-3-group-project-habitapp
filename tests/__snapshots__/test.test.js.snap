// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add_habit.js should render correctly 1`] = `undefined`;

exports[`Habit.js should render correctly 1`] = `
<Habit>
  <div>
    <p>
      <a
        href="http://localhost:8081/users/undefined/habits/undefined"
      />
      <p>
        frequency: 
         / 
      </p>
      <p>
        streak: 
      </p>
    </p>
  </div>
</Habit>
`;

exports[`Login.js should render correctly 1`] = `
<div>
  <h2
    className="app_title"
  >
    Habitapp
  </h2>
  <div
    className="login_box_outer"
  >
    <div
      className="login_box"
    >
      <div
        className="login_vert_box"
      >
        <form
          action="http://localhost:8081/authentication"
          method="POST"
        >
          <div
            className="form_component"
          >
            <label
              for="username"
            >
              Username:
            </label>
            <input
              name="username"
              type="text"
            />
          </div>
          <div
            className="form_component"
          >
            <label
              for="password"
            >
              Password:
            </label>
            <input
              name="password"
              type="password"
            />
          </div>
          <input
            className="submit_button"
            type="submit"
            value="submit"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Shallow Habits should render correctly 1`] = `
<div>
  <h2
    className="app_title"
  >
    Habitapp
  </h2>
  <div
    class="logout_button"
  >
    <Link
      to="/"
    >
      Click me to log out!
    </Link>
  </div>
  <p>
    Add a habit by 
    <Link
      to="/add_habit?id="
    >
      clicking here
    </Link>
  </p>
  <p>
    This will have a list of habits!
  </p>
  <p>
    didn't render
  </p>
</div>
`;
